posthog/models/kafka_partition_stats/sql.py

```diff
@@ -83,14 +94,69 @@
 """
 )
 
-CREATE_KAFKA_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS = CREATE_PARTITION_STATISTICS_KAFKA_TABLE(
-    KAFKA_EVENTS_PLUGIN_INGESTION
-)
-
 CREATE_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS_MV = CREATE_PARTITION_STATISTICS_MV(KAFKA_EVENTS_PLUGIN_INGESTION)
 
-DROP_KAFKA_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS = DROP_PARTITION_STATISTICS_KAFKA_TABLE(
-    KAFKA_EVENTS_PLUGIN_INGESTION
-)
-
 DROP_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS_MV = DROP_PARTITION_STATISTICS_MV(KAFKA_EVENTS_PLUGIN_INGESTION)
+
+# V2
+
+
+class PartitionStatsV2Table:
+    table_name: str = "events_plugin_ingestion_partition_statistics_v2"
+
+    def get_create_table_sql(self) -> str:
+        engine = ReplacingMergeTree(self.table_name, ver="timestamp")
+        return f"""
+            CREATE TABLE IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.{self.table_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}' (
+                topic LowCardinality(String),
+                partition UInt64,
+                offset UInt64,
+                token String,
+                distinct_id String,
+                ip String,
+                event String,
+                data_length UInt64,
+                timestamp DateTime
+            )
+            ENGINE = {engine}
+            PARTITION BY (topic, toStartOfDay(timestamp))
+            ORDER BY (topic, partition, offset)
+            TTL timestamp + INTERVAL 30 DAY
+        """
+
+    def get_drop_table_sql(self) -> str:
+        return f"""
+            DROP TABLE IF EXISTS `{CLICKHOUSE_DATABASE}`.{self.table_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}' SYNC
+        """
+
+
+@dataclass
+class PartitionStatsV2MaterializedView:
+    to_table: PartitionStatsV2Table
+    from_table: PartitionStatsKafkaTable
+
+    @property
+    def table_name(self) -> str:
+        return f"{self.from_table.topic}_partition_statistics_v2_mv"
+
+    def get_create_table_sql(self) -> str:
+        return f"""
+            CREATE MATERIALIZED VIEW IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.{self.table_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}'
+            TO `{CLICKHOUSE_DATABASE}`.{self.to_table.table_name}
+            AS SELECT
+                _topic AS topic,
+                _partition AS partition,
+                _offset AS offset,
+                token,
+                distinct_id,
+                ip,
+                if(startsWith(JSONExtractString(data, 'event') AS _event, '$'), _event, '') AS event,
```

Comment:
I don't think there's much reason to track everybody's custom events as those don't (or shouldn't) have much effect on ingestion performance.

Reply:
We've used it before to communicate with a customer that they are sending `xyz` event with only one distinct ID a million times a second. Basically saves them a bit of time looking into it themselves.

From a diagnostic perspective I agree, but from a comms perspective it has helped quite a bit. 'Please stop producing `xyz` event 🙏 '